src/goto-instrument/model_argc_argv.cpp

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************\
```

Comment:
It currently indeed is C specific, and the updated version includes an explicit check for mode==ID_C. I would, however, appreciate any work for supporting Java.